pyspecProcScripts/phasing.py

```diff
@@ -186,17 +186,56 @@ def costfun(ph1):
     # }}}
 
 
-def fid_from_echo(d, signal_pathway, fl=None, add_rising=False, fraction_nonnoise=0.1, direct="t2",
+def fid_from_echo(d, signal_pathway=None, fl=None, add_rising=False, direct="t2",
         exclude_rising=3,
         slice_multiplier=20,
         peak_lower_thresh=0.1,
         show_hermitian_sign_flipped=False,
         show_shifted_residuals=False):
+    """
+
+    Parameters
+    ==========
+    signal_pathway: dict
+        coherence transfer pathway that correspond to the signal
+    fl: figlist or None (default)
+        If you want the diagnostic plots (showing the distribution of the
+        data in the complex plane), set this to your figlist object.
+    add_rising: boolean
+        Take the first part of the echo (that which rises to the maximum)
+        and add the decaying (FID like) part. This increases the SNR of
+        the early points of the signal.
+    direct: string
+        Name of the direct dimension
+    exclude_rising: int
+        In general it is assumed that the first few points of signal
+        might be messed up due to dead time or other issues (assuming a
+        tau of 0).  This option allows us to add a rising edge to the
+        echo and exclude the first few points of the signal. Note, to use
+        this, add_rising must be True. 
+    slice_multiplier: int
+        The calculated frequency slice is calculated by taking the center
+        frequency and extending out to values that are included in the
+        peak times this multiplier. Therefore the larger this value the
+        larger the frequency slice. Increasing this value might serve
+        useful in the case of noisy spectra.
+    peak_lower_thresh: float
+        Fraction of the signal intensity used in calculating the
+        frequency slice. The smaller the value, the wider the slice. 
+    show_hermitian_sign_flipped: boolean
+        Diagnostic in checking the sign of the signal prior to the
+        hermitian phase correction
+    show_shifted_residuals: boolean
+        Diagnostic in analyzing the residuals after the hermitian phase
+        correction.
+
+    Returns                                 
+    =======
+    d: nddata
+        FID of properly sliced and phased signal
+    """
     # {{{ autodetermine slice range
     freq_envelope = d.C
-    freq_envelope.ift('t2')
-    freq_envelope = freq_envelope['t2':(0,None)] # slice out rising echo estimate according to experimental tau in order to limit oscillations
-    freq_envelope.ft('t2')
```

Comment:
see [thread](https://jmfrancklab.slack.com/archives/CV553TPRV/p1719501246917169) - this results in the rising time for some datasets to be thrown out when the correct time axis is applied